src/detect-engine-threshold.c

```diff
@@ -150,7 +150,7 @@ int ThresholdTimeoutCheck(Host *host, struct timeval *tv)
 
     prev = NULL;
     while (tmp != NULL) {
-        if ((tv->tv_sec - tmp->tv_sec1) <= tmp->seconds) {
+        if (abs(tv->tv_sec - tmp->tv_sec1) <= tmp->seconds) {
```

Comment:
AFAICS this threshold change has nothing to do with the flow timestamp change. Thus it should not be part of the same commit.